core/src/dbs/options.rs

```diff
@@ -339,13 +339,13 @@ impl Options {
 	}
 
 	/// Get currently selected NS
-	pub fn ns(&self) -> &str {
-		self.ns.as_ref().map(AsRef::as_ref).unwrap()
+	pub fn ns(&self) -> Result<&str, Error> {
+		self.ns.as_ref().map(AsRef::as_ref).ok_or(Error::Unreachable("Options::ns"))
```

Comment:
Don't we already have an error saying like "No namespace selected"?

Reply:
You are right! Considering that this is evidently not so "unreachable", I will update this error to be more specific and probably either mirror or replace the behavior of `valid_for_ns` and return `Error::NsEmpty`.